compiler/circle-interpreter-test/src/circle-interpreter.test.cpp

```diff
@@ -0,0 +1,265 @@
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <vector>
+
+class circle_interpreter_test : public ::testing::Test
+{
+protected:
+  bool initialize(void);
+  bool run(const std::string &command);
+
+protected:
+  bool compare(const std::string &file1, const std::string &file2);
+
+protected:
+  std::string _artifacts_path;
+  std::string _circle_interpreter_path;
+  std::string _result;
+};
+
+bool circle_interpreter_test::initialize(void)
+{
+  char *path = std::getenv("ARTIFACTS_PATH");
+  if (path == nullptr)
+  {
+    std::cerr << "ARTIFACTS_PATH not found" << std::endl;
+    return false;
+  }
+  _artifacts_path = path;
+
+  path = std::getenv("CIRCLE_INTERPRETER_PATH");
+  if (path == nullptr)
+  {
+    std::cerr << "ARTIFACTS_BIN_PATH not found" << std::endl;
+    return false;
+  }
+  _circle_interpreter_path = path;
+
+  return true;
+}
+
+bool circle_interpreter_test::run(const std::string &command)
+{
+  std::vector<char> buffer(260);
+  std::string result = "";
+  std::string cmd_err = command + " 2>&1";
+  FILE *pipe = popen(cmd_err.c_str(), "r");
+  if (!pipe)
+  {
+    return false;
+  }
+  try
+  {
+    while (fgets(&buffer[0], buffer.size(), pipe) != NULL)
+    {
+      result += &buffer[0];
+    }
+  }
+  catch (...)
+  {
+    pclose(pipe);
+    return false;
+  }
+  pclose(pipe);
+  _result = result;
+
+  std::cout << _result << std::endl;
+
+  return true;
+}
+
+bool circle_interpreter_test::compare(const std::string &file1, const std::string &file2)
+{
+  std::ifstream f1(file1.c_str(), std::ifstream::in | std::ifstream::binary);
+  std::ifstream f2(file2.c_str(), std::ifstream::in | std::ifstream::binary);
+
+  if (!f1.is_open() || !f2.is_open())
+  {
+    return false;
+  }
+
+  char *buffer1 = new char[1024]();
+  char *buffer2 = new char[1024]();
+
+  do
+  {
+    f1.read(buffer1, 1024);
+    f2.read(buffer2, 1024);
+
+    if (std::memcmp(buffer1, buffer2, 1024) != 0)
+    {
+      delete[] buffer1;
+      delete[] buffer2;
+      return false;
+    }
+  } while (f1.good() || f2.good());
+
+  delete[] buffer1;
+  delete[] buffer2;
+  return true;
+}
+
+TEST_F(circle_interpreter_test, show_help_msg)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string command = _circle_interpreter_path + " -h";
+  if (!run(command))
+  {
+    FAIL();
+    return;
+  }
+
+  const auto pos = _result.find("Usage: ./circle-interpreter");
+  ASSERT_NE(std::string::npos, pos);
+}
+
+TEST_F(circle_interpreter_test, valid_command)
+{
+  if (!initialize())
+  {
+    FAIL();
+    return;
+  }
+
+  std::string model = _artifacts_path + "/Conv2D_000.circle";
+  std::string input_prefix = _artifacts_path + "/Conv2D_000.circle.input";
+  std::string output_prefix = "/tmp/Conv2D_000.circle.output";
+  std::string generated_output = output_prefix + "0";
+  std::remove(generated_output.c_str());
```

Comment:
> output file will be left in `/tmp` directory. It would be better to remove it.
from https://github.com/Samsung/ONE/pull/9658#discussion_r970435174_

Before the test begins, clean the file name which will be generated. How about this approach? @jinevening

Reply:
Does `std::remove` remove the generated output file?

Reply:
Yes, it removes `generated_output` file if it exists before the test.

IMO, generated file under `/tmp` would not affect to others. (And also, I implemented this way since `circle-operator-test` used same way.)